### PR TITLE
Externalize strings

### PR DESCRIPTION
**Pull request by elharo**
@briandealwis and remove "Please" from UI messages

https://msdn.microsoft.com/en-us/library/windows/desktop/bb226825(v=vs.85).aspx

Avoid "please" except in situations where the user is asked to do something inconvenient or the software is to blame for the situation—for example, "Please wait while Windows copies the files to your computer."

### CODE REVIEW

File: plugins/com.google.cloud.tools.eclipse.appengine.newproject.maven/src/com/google/cloud/tools/eclipse/appengine/newproject/maven/MavenAppEngineStandardWizardPage.java [hunk 1]
```diff
@@ -261,26 +261,26 @@ private boolean validateGeneratedProjectLocation() {
...
       return false;
     } else if (!MavenCoordinatesValidator.validateArtifactId(artifactId)) {
-      setErrorMessage("Illegal Maven Artifact ID: " + artifactId);
+      setErrorMessage(Messages.getString("ILLEGAL_ARTIFACT_ID") + artifactId); //$NON-NLS-1$
       return false;
     }
     String version = getVersion();
     if (version.isEmpty()) {
-      setMessage("Please provide Maven artifact version.", INFORMATION);
+      setMessage(Messages.getString("PROVIDE_VERSION"), INFORMATION); //$NON-NLS-1$
```
Comment (briandealwis) [inline plugins/com.google.cloud.tools.eclipse.appengine.newproject.maven/src/com/google/cloud/tools/eclipse/appengine/newproject/maven/MavenAppEngineStandardWizardPage.java]: _Maven artifact version required_?

File: plugins/com.google.cloud.tools.eclipse.appengine.newproject.maven/src/com/google/cloud/tools/eclipse/appengine/newproject/maven/MavenAppEngineStandardWizardPage.java [hunk 2]
```diff
@@ -261,26 +261,26 @@ private boolean validateGeneratedProjectLocation() {
...
       return false;
     } else if (!MavenCoordinatesValidator.validateGroupId(groupId)) {
-      setErrorMessage(MessageFormat.format("Illegal Maven Group ID: {0}.", groupId));
+      setErrorMessage(MessageFormat.format(Messages.getString("ILLEGAL_GROUP_ID"), groupId)); //$NON-NLS-1$
       return false;
     }
     String artifactId = getArtifactId();
     if (artifactId.isEmpty()) {
-      setMessage("Please provide Maven Artifact ID.", INFORMATION);
+      setMessage(Messages.getString("PROVIDE_ARTIFACT_ID"), INFORMATION); //$NON-NLS-1$
```
Comment (briandealwis) [inline plugins/com.google.cloud.tools.eclipse.appengine.newproject.maven/src/com/google/cloud/tools/eclipse/appengine/newproject/maven/MavenAppEngineStandardWizardPage.java]: _Maven Artifact ID required_?

File: plugins/com.google.cloud.tools.eclipse.appengine.newproject.maven/src/com/google/cloud/tools/eclipse/appengine/newproject/maven/MavenAppEngineStandardWizardPage.java [hunk 3]
```diff
@@ -261,26 +261,26 @@ private boolean validateGeneratedProjectLocation() {
   private boolean validateMavenSettings() {
     String groupId = getGroupId();
     if (groupId.isEmpty()) {
-      setMessage("Please provide Maven Group ID.", INFORMATION);
+      setMessage(Messages.getString("PROVIDE_GROUP_ID"), INFORMATION); //$NON-NLS-1$
```
Comment (briandealwis) [inline plugins/com.google.cloud.tools.eclipse.appengine.newproject.maven/src/com/google/cloud/tools/eclipse/appengine/newproject/maven/MavenAppEngineStandardWizardPage.java]: _Maven Group ID required_?

File: plugins/com.google.cloud.tools.eclipse.appengine.newproject.maven/src/com/google/cloud/tools/eclipse/appengine/newproject/maven/MavenAppEngineStandardWizardPage.java [hunk 4]
```diff
@@ -227,7 +226,7 @@ public boolean validatePage() {
 
     String location = locationField.getText().trim();
     if (!useDefaults() && location.isEmpty()) {
-      setMessage("Please provide a location", INFORMATION);
+      setMessage(Messages.getString("PROVIDE_LOCATION"), INFORMATION); //$NON-NLS-1$
```
Comment (briandealwis) [inline plugins/com.google.cloud.tools.eclipse.appengine.newproject.maven/src/com/google/cloud/tools/eclipse/appengine/newproject/maven/MavenAppEngineStandardWizardPage.java]: Would _Missing location_ or _Location required_ make more sense then?

File: plugins/com.google.cloud.tools.eclipse.appengine.newproject.maven/src/com/google/cloud/tools/eclipse/appengine/newproject/maven/messages.properties [hunk 5]
```diff
@@ -5,3 +5,25 @@ APPENGINE_GUESTBOOK_ARCHETYPE_DESCRIPTION=Generates the guestbook demo sample, c
 APPENGINE_GUESTBOOK_ARCHETYPE_DISPLAY_NAME=Guestbook example
 APPENGINE_SKELETON_ARCHETYPE_DESCRIPTION=Generates a new, empty App Engine project ready for your own classes and resources, but with required files and directories.
 APPENGINE_SKELETON_ARCHETYPE_DISPLAY_NAME=Hello-world template
+LOCATION_GROUP_TEXT=Location
+JAVA_PACKAGE_LABEL=Java package:
+PROJECT_ID_LABEL=App Engine Project ID: (optional)
+GENERATED_PROJECT_LOCATION=Please select the location to contain generated project
```
Comment (briandealwis) [inline plugins/com.google.cloud.tools.eclipse.appengine.newproject.maven/src/com/google/cloud/tools/eclipse/appengine/newproject/maven/messages.properties]: Missed _Please_. _Select the location…_ sounds ok.

Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.appengine.newproject.maven/src/com/google/cloud/tools/eclipse/appengine/newproject/maven/messages.properties]: done

File: plugins/com.google.cloud.tools.eclipse.appengine.newproject.maven/src/com/google/cloud/tools/eclipse/appengine/newproject/maven/MavenAppEngineStandardWizardPage.java (hunk 1 above)
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.appengine.newproject.maven/src/com/google/cloud/tools/eclipse/appengine/newproject/maven/MavenAppEngineStandardWizardPage.java]: ditto

File: plugins/com.google.cloud.tools.eclipse.appengine.newproject.maven/src/com/google/cloud/tools/eclipse/appengine/newproject/maven/MavenAppEngineStandardWizardPage.java (hunk 2 above)
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.appengine.newproject.maven/src/com/google/cloud/tools/eclipse/appengine/newproject/maven/MavenAppEngineStandardWizardPage.java]: ditto

File: plugins/com.google.cloud.tools.eclipse.appengine.newproject.maven/src/com/google/cloud/tools/eclipse/appengine/newproject/maven/messages.properties [hunk 6]
```diff
@@ -5,3 +5,25 @@ APPENGINE_GUESTBOOK_ARCHETYPE_DESCRIPTION=Generates the guestbook demo sample, c
 APPENGINE_GUESTBOOK_ARCHETYPE_DISPLAY_NAME=Guestbook example
 APPENGINE_SKELETON_ARCHETYPE_DESCRIPTION=Generates a new, empty App Engine project ready for your own classes and resources, but with required files and directories.
 APPENGINE_SKELETON_ARCHETYPE_DISPLAY_NAME=Hello-world template
+LOCATION_GROUP_TEXT=Location
```
Comment (akerekes) [inline plugins/com.google.cloud.tools.eclipse.appengine.newproject.maven/src/com/google/cloud/tools/eclipse/appengine/newproject/maven/messages.properties]: In many of the other modules we use lowercase dot separated format, e.g.: `location.group.text`. Clearly not a critical issue, but an inconsistency.

**Pull request merged**